csharp/test/Apache.Arrow.Tests/StringArrayTests.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Xunit;
+
+namespace Apache.Arrow.Tests
+{
+    public class StringArrayTests
+    {
+        public class GetString
+        {
+            [Theory]
+            [InlineData(null, null)]
+            [InlineData(null, "")]
+            [InlineData(null, "value")]
+            [InlineData("", null)]
+            [InlineData("", "")]
+            [InlineData("", "value")]
+            [InlineData("value", null)]
+            [InlineData("value", "")]
+            [InlineData("value", "value")]
+            public void ReturnsAppendedValue(string firstValue, string secondValue)
+            {
+                // Arrange
+                var array = new StringArray.Builder()
+                    .Append(firstValue)
+                    .Append(secondValue)
```

Comment:
What's the purpose of `secondValue` here?  I'm guessing this is to reproduce the original issue?  Can you add a comment to this effect so that others understand the purpose of this test?

Reply:
Correct, it demonstrates that additional values in the array can influence the value returned by GetString (at least prior to this change). I'll add a comment.

Reply:
Hi @westonpace, I wanted to bring attention that I added a comment to the test per your review comments when you have a chance to look at it again.